deployment_scripts/linode-marketplace-nats-single-node/nats-single-node-deploy.sh

```diff
@@ -0,0 +1,147 @@
+#<UDF name="soa_email_address" label="Email address (for the Let's Encrypt SSL certificate)" example="user@domain.tld">
+# <UDF name="nats_port" label="NATS Server Port" default="4222" />
+# <UDF name="websocket_port" label="NATS Websocket Port" default="8888" />
+# <UDF name="mqtt_port" label="NATS MQTT Port" default="1883" />
+
+
+# git repo
+export GIT_REPO="https://github.com/jcotoBan/marketplace-apps.git"
+export WORK_DIR="/tmp/marketplace-apps" 
+export MARKETPLACE_APP="apps/linode-marketplace-nats-single-node"
+
+# enable logging
+exec > >(tee /dev/ttyS0 /var/log/stackscript.log) 2>&1
+
+function cleanup {
+  if [ -d "${WORK_DIR}" ]; then
+    rm -rf ${WORK_DIR}
+  fi
+
+}
+
+function udf {
+  local group_vars="${WORK_DIR}/${MARKETPLACE_APP}/group_vars/linode/vars"
+  echo "webserver_stack: lemp" >> ${group_vars};
+  
+  if [[ -n ${USER_NAME} ]]; then
+    echo "username: ${USER_NAME}" >> ${group_vars};
+  else echo "No username entered";
+  fi
+
+  if [ "$DISABLE_ROOT" = "Yes" ]; then
+    echo "disable_root: yes" >> ${group_vars};
+  else echo "Leaving root login enabled";
+  fi
+
+  if [[ -n ${PASSWORD} ]]; then
+    echo "password: ${PASSWORD}" >> ${group_vars};
+  else echo "No password entered";
+  fi
+
+  if [[ -n ${PUBKEY} ]]; then
+    echo "pubkey: ${PUBKEY}" >> ${group_vars};
+  else echo "No pubkey entered";
+  fi
+
+  if [[ -n ${SOA_EMAIL_ADDRESS} ]]; then
+    echo "soa_email_address: ${SOA_EMAIL_ADDRESS}" >> ${group_vars};
+  fi
+
+  if [[ -n ${DOMAIN} ]]; then
+    echo "domain: ${DOMAIN}" >> ${group_vars};
+  else
+    echo "default_dns: $(hostname -I | awk '{print $1}'| tr '.' '-' | awk {'print $1 ".ip.linodeusercontent.com"'})" >> ${group_vars};
+  fi
+
+
+  if [[ -n ${SUBDOMAIN} ]]; then
+    echo "subdomain: ${SUBDOMAIN}" >> ${group_vars};
+  else 
+    echo "subdomain: www" >> ${group_vars};
+  fi
+
+
+  #nats vars
+  
+  if [[ -n ${NAME} ]]; then
+    echo "name: ${NAME}" >> ${group_vars};
+  fi
+
+  if [[ -n ${VERSION} ]]; then
+    echo "version: ${VERSION}" >> ${group_vars};
+  fi
+
+  if [[ -n ${SYSTEM_USER_PASSWORD} ]]; then
+    echo "system_user_password: ${SYSTEM_USER_PASSWORD}" >> ${group_vars};
+  fi
+
+  if [[ -n ${EXAMPLE_USER_PASSWORD} ]]; then
+    echo "example_user_password: ${EXAMPLE_USER_PASSWORD}" >> ${group_vars};
+  fi
+
+  if [[ -n ${NATS_PORT} ]]; then
+    echo "nats_port: ${NATS_PORT}" >> ${group_vars};
+  fi
+
+  if [[ -n ${WEBSOCKET_PORT} ]]; then
+    echo "websocket_port: ${WEBSOCKET_PORT}" >> ${group_vars};
+  fi
+
+  if [[ -n ${MQTT_PORT} ]]; then
+    echo "mqtt_port: ${MQTT_PORT}" >> ${group_vars};
+  fi
+
+}
+
+function run {
+  # install dependancies
+  apt-get update
+  apt-get install -y git python3 python3-pip
+
+  # clone repo and set up ansible environment
+  git -C /tmp clone -b nats-single-node ${GIT_REPO} #to test, change accord.
+  # for a single testing branch
+  # git -C /tmp clone --single-branch --branch ${BRANCH} ${GIT_REPO}
```

Comment:
change to match the other apps:
`# git -C /tmp clone -b ${BRANCH} ${GIT_REPO}`

Reply:
Fixed!